mediar-ai/screenpipe
# Update xcap to better support capturing screenhots on Wayland

---
name: Update xcap
about: submit changes to the project
title: "[Update xcap, to include important bugfixes, and better support Wayland.] "
labels: ''bugfix"
assignees: ''"

---

## description
This PR updates the xcap version to the newest one in the `xcap` repo. The newest version of `xcap` contains important fixes for Wayland. 

related issue: #63 

## type of change

## how to test
1.  Run screenpipe on Wayland
2. Check that it takes screnshoots(DE dependent, conformed to work on Gnome)
3. Check that there are no warning messages
4. Run the project for a couple of minutes to see that it no longer segfaults


## checklist
- [X] i have read the [CONTRIBUTING.md](../CONTRIBUTING.md) file 
- [X] i have added tests that prove my fix is effective or that my feature works - *the tests are in the `xcap` repo, since the original bug was there*

## additional notes
Depending on your dependency policy, you might want to wait till a new version of xcap is released to `crates.io`. Cargo does not allow a crate to have different sources depending on the target, so updating `xcap` requires updating it for all the targets. This is something you might not want to do.

In such a case, this PR can serve as a way to test out the updates to `xcap`,  before a new version of the crate release.
/claim #63
